File: Arrays/1313. Decompress Run-Length Encoded List.js
link = 'https://leetcode.com/problems/decompress-run-length-encoded-list/'


// Solution
//---------------------------------
const decompressRLElist = (nums) => {
    let output = []
    // let freqVal = []
    for (let i = 0; i < nums.length; i=i+2) {
        // let obj = {
        //     freq : nums[i],
        //     value : nums[i+1]
        // }
        // freqVal.push(obj)
        for (let j = 0; j < nums[i]; j++) {
            output.push(nums[i+1])
        }
    }
    return output
}
//---------------------------------
let result = decompressRLElist([1,2,3,4])
console.log(result)

File: Arrays/1313. Decompress Run-Length Encoded List.py
def decompressRLElist(nums):
    output = []
    for i in range(0, len(nums),2):
        for j in range(0, nums[i]):
            output.append(nums[i+1])
    return output

print(decompressRLElist([1,2,3,4]))